unittests/backends/QPPTester.cpp

```diff
@@ -1242,3 +1242,211 @@ CUDAQ_TEST(QPPTester, checkReset) {
     EXPECT_EQ(1, qppBackend.mz(q1));
   }
 }
+
+CUDAQ_TEST(QPPTester, checkStateAllocation) {
+
+  {
+    // Initialize QPP Backend with 1 qubit.
+    const int qubit_count = 1;
+    QppCircuitSimulator<qpp::ket> qppBackend;
+
+    // Build up the state as a vector of doubles, then pass
+    // that along to the `allocateQubits` function and ensure
+    // it returns the expected state.
+    std::vector<std::complex<double>> inputState(pow(2, qubit_count), M_PI);
+    auto qubits = qppBackend.allocateQubits(qubit_count, inputState.data());
+
+    qpp::ket got_state = qppBackend.getStateVector();
+    qpp::ket want_state = getZeroState(qubit_count);
+    want_state.fill(M_PI);
+    EXPECT_EQ(want_state, got_state);
+  }
+
+  {
+    // Initialize QPP Backend with 2 qubits.
+    const int qubit_count = 2;
+    QppCircuitSimulator<qpp::ket> qppBackend;
+
+    // Build up the state as a vector of doubles, then pass
+    // that along to the `allocateQubits` function and ensure
+    // it returns the expected state.
+    std::vector<std::complex<double>> inputState(pow(2, qubit_count), M_PI_2);
+    auto qubits = qppBackend.allocateQubits(qubit_count, inputState.data());
+
+    qpp::ket got_state = qppBackend.getStateVector();
+    qpp::ket want_state = getZeroState(qubit_count);
+    want_state.fill(M_PI_2);
+    EXPECT_EQ(want_state, got_state);
+  }
+
+  {
+    // Initialize QPP Backend with 3 qubits.
+    const int qubit_count = 3;
+    QppCircuitSimulator<qpp::ket> qppBackend;
+
+    // Build up the state as a vector of doubles, then pass
+    // that along to the `allocateQubits` function and ensure
+    // it returns the expected state.
+    std::vector<std::complex<double>> inputState(pow(2, qubit_count), M_PI_4);
+    auto qubits = qppBackend.allocateQubits(qubit_count, inputState.data());
+
+    qpp::ket got_state = qppBackend.getStateVector();
+    qpp::ket want_state = getZeroState(qubit_count);
+    want_state.fill(M_PI_4);
+    EXPECT_EQ(want_state, got_state);
+  }
+
+  // Broken integration test with sampling and measurements.
```

Comment:
Excuse the mess in the rest of this PR, you could cut and paste just this test into the head features/qalloc branch and not need to pull down the rest of this code. 

We have a dimension mismatch issue when intermixing conventional qubit allocations with state vector allocations. The dimension error only shows up *after* the state allocation, but prevents *any* further state manipulation. I can reproduce the error with a sample call, measure call, and any gate application (qpp::apply). I'm sure there are more, but it isn't fully clear to me where the dimension issues come from. We are only using free QPP functions, no classes that should be retaining any sort of state dimension -- that should come directly from the state vector itself. And that state vector that I'm printing out looks as I'd expect it to.